### PR TITLE
Make default onError log the error with a debug level

### LINKED ISSUE #58: Make default onError log the error with a debug levle
Let's change this default implementation. It would significantly better if by default we logged this error as `debug`.

_Originally posted by @mcollina in https://github.com/mercurius-js/cache/pull/48#discussion_r783975298_

### PR DESCRIPTION
**Pull request by jhonrocha**
Uses `onError = app.log.debug`, if available.

Closes #58

### CODE REVIEW

File: lib/validation.js [hunk 1]
```diff
@@ -64,6 +64,8 @@ function validateOpts (app, opts = {}) {
     if (typeof onError !== 'function') {
       throw new Error('onError must be a function')
     }
+  } else if (app.log && app.log.debug && typeof app.log.debug === 'function') {
+    onError = app.log.debug
```
Comment (mcollina) [inline lib/validation.js:68]: I'm sure this would work as expected with pino, check it out.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: lib/validation.js [hunk 2]
```diff
@@ -64,6 +64,10 @@ function validateOpts (app, opts = {}) {
     if (typeof onError !== 'function') {
       throw new Error('onError must be a function')
     }
+  } else if (app.log && app.log.debug && typeof app.log.debug === 'function') {
+    onError = (prefix, fieldName, err) => {
+      app.log.debug({ err, msg: 'Mercurius cache error', prefix, fieldName })
+    }
```
Comment (mcollina) [inline lib/validation.js:70]: Can you please generate this closure in a top level function? So we can avoid closing over opts.

Comment (jhonrocha) [reply lib/validation.js]: Is a function factory the best option in this case? You can see it on the PR now:
```
function defaultDebug (app) {
  return function (prefix, fieldName, err) {
    app.log.debug({ err, msg: 'Mercurius cache error', prefix, fieldName })
  }
}
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**